### ISSUE TITLE
Update dashboard permission tab to meet accessibility requirements

### ISSUE THREAD

**Issue by IevaVasiljeva** [grafana/grafana#77203]
**What**

Make small tweaks to improve the accessibility of the dashboard permissions page.

**Why**

To make Grafana accessible.

**Context**

We've had to raise the accessibility issue threshold as part of https://github.com/grafana/grafana/pull/77155, as tests are failing when provisioned dashboard permissions are correctly set. It would be good to lower the threshold again. Slack thread: https://raintank-corp.slack.com/archives/CHKLEJ668/p1698311401063749

Problems reported by the tests:

```
Errors in http://grafana-server:3001/d/O6f11TZWk/panel-tests-bar-gauge?orgId=1&editview=permissions:

 • Buttons must have discernible text
   (https://dequeuniversity.com/rules/axe/4.4/button-name?application=axeAPI)

   (#pageContent > div:nth-child(4) > div > div:nth-child(1) > div >
   div:nth-child(2) > div > div:nth-child(3) > table > tbody > tr:nth-child(1)
   > td:nth-child(6) > button)

   <button class="css-72lnkn-button" type="button" tabindex="0"><svg
   xmlns="http://www.w3.org/2...</button>

 • Buttons must have discernible text
   (https://dequeuniversity.com/rules/axe/4.4/button-name?application=axeAPI)

   (#pageContent > div:nth-child(4) > div > div:nth-child(1) > div >
   div:nth-child(2) > div > div:nth-child(3) > table > tbody > tr:nth-child(2)
   > td:nth-child(6) > button)

   <button class="css-72lnkn-button" type="button" tabindex="0"><svg
   xmlns="http://www.w3.org/2...</button>

 • Buttons must have discernible text
   (https://dequeuniversity.com/rules/axe/4.4/button-name?application=axeAPI)

   (#pageContent > div:nth-child(4) > div > div:nth-child(1) > div >
   div:nth-child(2) > div > div:nth-child(3) > table > tbody > tr:nth-child(3)
   > td:nth-child(6) > button)

   <button class="css-72lnkn-button" type="button" tabindex="0"><svg
   xmlns="http://www.w3.org/2...</button>

 • IDs of active elements must be unique
   (https://dequeuniversity.com/rules/axe/4.4/duplicate-id-active?application=axeAPI)

   (#Layer_1)

   <svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg"
   viewBox="0 0 24 24" width="16" height="16" class="css-1d3xu67-Icon"
   tabindex="0"><path d="M12,2A10,10,0,1,0,22,1...</svg>

 • Elements must only use allowed ARIA attributes
   (https://dequeuniversity.com/rules/axe/4.4/aria-allowed-attr?application=axeAPI)

   (#pageContent > div:nth-child(4) > div > div:nth-child(1) > div >
   div:nth-child(2) > div > div:nth-child(2) > div)

   <div class="cta-form" aria-label="Permissions slider"><button
   aria-label="Close" clas...</div>

 • Elements must have sufficient color contrast
   (https://dequeuniversity.com/rules/axe/4.4/color-contrast?application=axeAPI)

   (#pageContent > div:nth-child(4) > div > div:nth-child(1) > div >
   div:nth-child(2) > div > div:nth-child(3) > table > thead > tr >
   th:nth-child(2))

   <th>Role</th>

 • Elements must have sufficient color contrast
   (https://dequeuniversity.com/rules/axe/4.4/color-contrast?application=axeAPI)

   (#pageContent > div:nth-child(4) > div > div:nth-child(1) > div >
   div:nth-child(2) > div > div:nth-child(3) > table > thead > tr >
   th:nth-child(4))

   <th style="width: 40%;">Permission</th>

 • Elements must have sufficient color contrast
   (https://dequeuniversity.com/rules/axe/4.4/color-contrast?application=axeAPI)

   (#pageContent > div:nth-child(4) > div > div:nth-child(1) > div >
   div:nth-child(2) > div > div:nth-child(3) > table > tbody > tr:nth-child(1)
   > td:nth-child(2) > span)

   <span>Admin </span>

 • Elements must have sufficient color contrast
   (https://dequeuniversity.com/rules/axe/4.4/color-contrast?application=axeAPI)

   (#pageContent > div:nth-child(4) > div > div:nth-child(1) > div >
   div:nth-child(2) > div > div:nth-child(3) > table > tbody > tr:nth-child(2)
   > td:nth-child(2) > span)

   <span>Editor </span>

 • Elements must have sufficient color contrast
   (https://dequeuniversity.com/rules/axe/4.4/color-contrast?application=axeAPI)

   (#pageContent > div:nth-child(4) > div > div:nth-child(1) > div >
   div:nth-child(2) > div > div:nth-child(3) > table > tbody > tr:nth-child(2)
   > td:nth-child(3) > em)

   <em class="muted no-wrap">Inherited from folder</em>
```

**Comment by grafanabot** [comment]
@IevaVasiljeva please add one or more appropriate labels. Here are some tips:

- if you are making an issue, TODO, or reminder for yourself or your team, please add one label that best describes the product or feature area. Please also add the issue to your project board. :rocket:

- if you are making an issue for any other reason (docs typo, you found a bug, etc), please add at least one label that best describes the product or feature that you are discussing (e.g. `area/alerting`, `datasource/loki`, `type/docs`, `type/bug`, etc). [Our issue triage](https://github.com/grafana/grafana/blob/main/ISSUE_TRIAGE.md#3-categorizing-an-issue) doc also provides additional guidance on labeling. :rocket:

 Thank you! :heart:

**Comment by mrkirthi-24** [comment]
Hey @IevaVasiljeva I would like to work on this issue

**Comment by IevaVasiljeva** [comment]
That's great @mrkirthi-24, go for it!

**Comment by joshhunt** [comment]
We've changed this view a bit (it's now a drawer instead of a page, and probably no longer visited by pa11y), so we'll need to investigate this a bit.

**Comment by eledobleefe** [comment]
I cannot use `pa11y` on my computer but I investigated it using "Accessibility Insights for Web", and there still seems to be some a11y issues.
<img width="1146" alt="Captura de pantalla 2024-02-28 a las 17 53 16" src="https://github.com/grafana/grafana/assets/65417731/5e24e4e9-0093-427a-bd3d-5e041fa2c08c">
_Note: in the screenshot, there are some checks failing that belong to the page, not the permissions drawer_

**Comment by yaelleC** [comment]
@grafana/sharing-squad we've moved this to your board seeing as you own the dashboard permission UX

**Comment by IevaVasiljeva** [comment]
I've checked, and the threshold for dash permission related a11y errors and warnings has been lowered to 5 in https://github.com/grafana/grafana/pull/85178. This is even lower than where it was initially (9), so I think we're good to close this issue. Thanks for the ping.
